File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    let cl = Array.from({length : n+2}, (val,idx) => idx === 0 || idx === n+1 ? 0 : 1)
    lost.forEach((val) => cl[val] = 0)
    reserve.forEach((val) => cl[val] += 1)
    
    let cnt = 0
    for (let i = 1; i <= n ; i++) {
        if (cl[i] === 0){
            if(cl[i-1] === 2){
                cl[i-1] -= 1
                cl[i] += 1
            }
            else if(cl[i+1] === 2){
                cl[i+1] -= 1
                cl[i] += 1
            }
        }
        if (cl[i] >= 1) {
            cnt += 1
        }
    }
    return cnt;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
# 2 <= n <= 30
# 1 <= lost <= n
# 1 <= rese <= n

def solution(n, lost, reserve):
    CL = [0] + [1] * n + [0]
    cnt = 0
    for i in range(1,n + 1):
        if i in lost : CL[i] -= 1
        if i in reserve : CL[i] += 1
    
    for i in range(1,n + 1):
        if CL[i] == 0:
            if CL[i-1] == 2: CL[i-1], CL[i] = CL[i-1] - 1, CL[i] + 1
            elif CL[i+1] == 2: CL[i+1], CL[i] = CL[i+1] -1 , CL[i] + 1        
        if CL[i] : cnt += 1    
    
    return cnt
    
